core/templates/pages/learner-dashboard-page/goals-tab.component.ts

```diff
@@ -82,13 +84,14 @@ export class GoalsTabComponent implements OnInit {
   editGoalsTopicClassification: number[] = [];
   editGoalsTopicBelongToLearntToPartiallyLearntTopic: boolean[] = [];
   windowIsNarrow: boolean = false;
-  showThreeDotsDropdown: boolean = false;
   directiveSubscriptions = new Subscription();
 
   ngOnInit(): void {
     this.MAX_CURRENT_GOALS_LENGTH = AppConstants.MAX_CURRENT_GOALS_COUNT;
     this.currentGoalsStoryIsShown = [];
+    this.showThreeDotsDropdown = [];
     this.currentGoalsStoryIsShown[0] = true;
+    this.showThreeDotsDropdown[i] = false;
```

Comment:
I think you would have to initialize the array to be as long as the number of goals, and set all the elements to be false (for that, again, your would have to run a loop from 0 (inclusive) through to the total number of goals (not inclusive)).

Reply:
Done!